scripts/target_determinator.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+"""Computes the potentially differing rules from some git commit.
+
+Wraps the "target-determinator" Go program here:
+https://github.com/bazel-contrib/target-determinator
+
+The purpose is to compute the potentially impacted set of targets from some
+provided Git commit to the current checkout.
+
+This script will ensure a cached version of the latest release is available, and
+then forward a limited set of flags to it. This script also filters the
+resulting targets using `bazel query` to make it the most relevant list for
+continuous integration.
+"""
+
+__copyright__ = """
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""
+
+import subprocess
+import argparse
+import tempfile
+import sys
+from pathlib import Path
+from typing import List
+
+import scripts_utils
+
+
+def log(s: str) -> None:
+    print(s, file=sys.stderr)
+
+
+def quiet_run_output(args: List[str]) -> str:
+    try:
+        p = subprocess.run(
+            args,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            check=True,
+            encoding="utf-8",
+        )
+        return p.stdout
+    except subprocess.CalledProcessError as err:
+        log(err.stderr)
+        raise
+
+
+def filter_targets(bazel: Path, targets: str) -> str:
+    with tempfile.NamedTemporaryFile(mode="w+") as tmp:
+        tmp.write(
+            f"let t = set({targets}) in "
+            "kind(rule, $t) except attr(tags, manual, $t)\n"
+        )
+        tmp.seek(0)
+        tmp_head = "".join(line for (line, _) in zip(tmp, range(10)))
+        if tmp.read(1) != "":
+            tmp_head += "...\n"
+        log(f"Bazel query file's first 10 lines:\n{tmp_head}---")
```

Comment:
This code is pretty confusing to me; if you want logging, can you log the query string directly?

Reply:
Sure, I'll switch to logging the string before writing it.